report/sql_tasks/tasks/report/create_from_scratch/04_hubspot_write/01_company_property_update/01_create_view.sql

```diff
@@ -0,0 +1,80 @@
+DROP VIEW IF EXISTS hubspot.company_property_update CASCADE;
+
+-- This view recreates the properties that we update on Hubspot companies
+CREATE VIEW hubspot.company_property_update AS (
+    WITH
+        metrics AS (
+            SELECT
+                column1 AS metric_name,
+                column2 AS role,
+                column3 AS timescale,
+                column4 AS offset
+            FROM (
+                VALUES
+                    ('lms_teachers_last_semester', 'teacher', 'semester', 1),
+                    ('lms_teachers_this_semester', 'teacher', 'semester', 0),
+                    ('lms_users_last_academic_year', 'user', 'academic_year', 1),
+                    ('lms_users_this_academic_year', 'user', 'academic_year', 0),
+                    ('lms_users_last_semester', 'user', 'semester', 1),
+                    ('lms_users_this_semester', 'user', 'semester', 0),
+                    ('lms_users_all_time', 'user', 'all_time', 0)
+            ) AS data
+        ),
+
+        values_in_rows AS (
+            SELECT
+                activity.organization_id,
+                metrics.metric_name,
+                activity.billable as count
+            FROM metrics
+
+            -- Get the correct time period to join on to
+            LEFT JOIN LATERAL (
+                SELECT DISTINCT(period)
+                FROM organization_activity
+                WHERE timescale::text = metrics.timescale
+                ORDER BY period DESC
+                LIMIT 1
+                OFFSET metrics.offset
+            ) metric_period ON TRUE
+
+            -- Get the correct metric row
+            LEFT OUTER JOIN LATERAL (
+                SELECT *
+                FROM organization_activity
+                WHERE
+                    timescale::text = metrics.timescale
+                    AND role::text = metrics.role
+                    AND period = metric_period.period
+            ) activity ON TRUE
+        ),
+
+        values_in_cols AS (
+            SELECT
+                organization_id,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_teachers_last_semester'), 0) AS lms_teachers_last_semester,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_teachers_this_semester'), 0) AS lms_teachers_this_semester,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_users_last_academic_year'), 0) AS lms_users_last_academic_year,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_users_this_academic_year'), 0) AS lms_users_this_academic_year,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_users_last_semester'), 0) AS lms_users_last_semester,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_users_this_semester'), 0) AS lms_users_this_semester,
+                COALESCE(MAX(count) FILTER (WHERE metric_name='lms_users_all_time'), 0) AS lms_users_all_time
+            FROM values_in_rows
+            GROUP BY organization_id
```

Comment:
We could have dealt with the metrics as columns in the Python side, but a model where you create a view with exactly what you want, then push it verbatim to Hubspot seems quite nice and verifiable.

We might want to deploy this PR in two chunks if we did it. To create the view, and then to use it. This would allow people to vet the numbers.